examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1190 @@
+      const std::vector<Tensor<1, dim>> &normals =
+        scratch_data.get_normal_vectors();
+
+      std::vector<double> g(n_q_points);
+      boundary_function.value_list(q_points, g);
+
+
+      const double extent1 = cell->extent_in_direction(
+        GeometryInfo<dim>::unit_normal_direction[face_no]);
+      const double penalty = compute_penalty(extent1, extent1);
+
+      for (unsigned int point = 0; point < n_q_points; ++point)
+        {
+          for (unsigned int i = 0; i < dofs_per_cell; ++i)
+            for (unsigned int j = 0; j < dofs_per_cell; ++j)
+              copy_data.cell_matrix(i, j) +=
+                (
+                  // - nu (\nabla u . n) v
+                  -viscosity * (fe_fv.shape_grad(j, point) * normals[point]) *
+                    fe_fv.shape_value(i, point)
+
+                  // - nu u (\nabla v . n)  // NIPG: use +
+                  - viscosity * fe_fv.shape_value(j, point) *
+                      (fe_fv.shape_grad(i, point) * normals[point])
+
+                  // + nu * penalty u v
+                  + viscosity * penalty * fe_fv.shape_value(j, point) *
+                      fe_fv.shape_value(i, point)) *
+                JxW[point];
+
+          for (unsigned int i = 0; i < dofs_per_cell; ++i)
+            copy_data.cell_rhs(i) +=
+              (
+                // -nu g (\nabla v . n) // NIPG: use +
+                -viscosity * g[point] *
+                  (fe_fv.shape_grad(i, point) * normals[point])
+
+                // +nu penalty g v
+                +
+                viscosity * penalty * g[point] * fe_fv.shape_value(i, point)) *
+              JxW[point];
+        }
+    };
+
+    auto face_worker = [&](const Iterator &    cell,
+                           const unsigned int &f,
+                           const unsigned int &sf,
+                           const Iterator &    ncell,
+                           const unsigned int &nf,
+                           const unsigned int &nsf,
+                           ScratchData &       scratch_data,
+                           CopyData &          copy_data) {
+      const FEInterfaceValues<dim> &fe_iv =
+        scratch_data.reinit(cell, f, sf, ncell, nf, nsf);
+
+      const auto &       q_points   = fe_iv.get_quadrature_points();
+      const unsigned int n_q_points = q_points.size();
+
+      copy_data.face_data.emplace_back();
+      CopyDataFace &     copy_data_face = copy_data.face_data.back();
+      const unsigned int n_dofs_face    = fe_iv.n_current_interface_dofs();
+      copy_data_face.joint_dof_indices  = fe_iv.get_interface_dof_indices();
+      copy_data_face.cell_matrix.reinit(n_dofs_face, n_dofs_face);
+
+      const std::vector<double> &        JxW     = fe_iv.get_JxW_values();
+      const std::vector<Tensor<1, dim>> &normals = fe_iv.get_normal_vectors();
+
+      const double extent1 =
+        cell->extent_in_direction(GeometryInfo<dim>::unit_normal_direction[f]) *
+        (cell->has_children() ? 2.0 : 1.0);
```

Comment:
Do we need this logic? We always assemble on active cells, so has_children()==false.